### ISSUE TITLE
The request could not be completed because: Unauthorized Access

### ISSUE THREAD

**Issue by assembleMHN** [fastlane/fastlane#20078]
When using fastlane to try to upload a app update it fails with the below response:

The request could not be completed because:	Unauthorized Access

Fastlane can communicate with the Apple Developer Portal with no issues, but every-time i need to do work in App Store Connect i get the above error message.

I have tried to delete the  /Users/username/.fastlane/spaceship/{email}/cookie that did not do anything, i also tried to change my appleID password and sign-out of all devices, that also did not do anything, which is the solution stated in the past issues i have been able to locate.

Below is the output of Fastlane when it fails:

[14:05:24]: Running lane `ios create_appstore_all`. Next time you can do this by directly typing `fastlane ios create_appstore_all` 🚀.
[14:05:24]: Driving the lane 'ios create_appstore_all' 🚀
[14:05:24]: --------------------------------------
[14:05:24]: --- Disse er envfiler bliver kørt: ---
[14:05:24]: --------------------------------------
[14:05:24]: ▸ xxxxxxxxxxxxxx
[14:05:24]: ---------------------------------------------
[14:05:24]: --- Step: ruby ./assets/GenerateManual.rb ---
[14:05:24]: ---------------------------------------------
[14:05:24]: $ ruby ./assets/GenerateManual.rb
[14:05:24]: ▸ Opdaterer manual
[14:05:24]: ----------------------------------------------------------
[14:05:24]: --- Step: Switch to ios set_environment_variables lane ---
[14:05:24]: ----------------------------------------------------------
[14:05:24]: Cruising over to lane 'ios set_environment_variables' 🚖
[14:05:24]: Cruising back to lane 'ios create_appstore_all' 🚘
[14:05:24]: ------------------------------------------------
[14:05:24]: --- Step: Switch to ios create_appstore lane ---
[14:05:24]: ------------------------------------------------
[14:05:24]: Cruising over to lane 'ios create_appstore' 🚖
[14:05:24]: xxxxxxxxxxxxxxxxxxxxxxx
[14:05:24]: xxxxxxxxxxxxxxxxxxxxxxx
[14:05:24]: xxxxxxxxxxxxxxxxxxxxxxx
[14:05:24]: ---------------------
[14:05:24]: --- Step: produce ---
[14:05:24]: ---------------------

+----------------+------------------------------------------------------------+
|                         Summary for produce 2.204.3                         |
+----------------+------------------------------------------------------------+
| app_identifier | xxxxxxxxxxxxxxxxxxxxxxx                       |
| sku            | xxxxxxxxxxxxxxxxxxxxxxx                       |
| app_name       | xxxxxxxxxxxxxxxxxxxxxxx                                              |
| language       | da                                                         |
| app_version    | 1.0                                                        |
| company_name   | xxxxxxxxxxxxxxxxxxxxxxx. |
| username       | xxxxxxxxxxxxxxxxxxxxxxx                                    |
| platform       | ios                                                        |
| skip_itc       | false                                                      |
| skip_devcenter | false                                                      |
| team_id        | xxxxxxxxxxxxxxxxxxxxxxx                                                 |
| itc_team_id    | xxxxxxxxxxxxxxxxxxxxxxx                                                  |
+----------------+------------------------------------------------------------+

[14:05:28]: [DevCenter] App 'xxxxxxxxxxxxxxxxxxxxxxx' already exists, nothing to do on the Dev Center
[14:05:57]: --------------------------
[14:05:57]: --- Step: notification ---
[14:05:57]: --------------------------
+------------------+-------------------------+
|                Lane Context                |
+------------------+-------------------------+
| DEFAULT_PLATFORM | ios                     |
| PLATFORM_NAME    | ios                     |
| LANE_NAME        | ios create_appstore_all |
+------------------+-------------------------+
[14:05:57]: Unauthorized Access

+------+----------------------------------------------+-------------+
|                         fastlane summary                          |
+------+----------------------------------------------+-------------+
| Step | Action                                       | Time (in s) |
+------+----------------------------------------------+-------------+
| 1    | Verifying fastlane version                   | 0           |
| 2    | default_platform                             | 0           |
| 3    | ruby ./assets/GenerateManual.rb              | 0           |
| 4    | Switch to ios set_environment_variables lane | 0           |
| 5    | Switch to ios create_appstore lane           | 0           |
| 💥   | produce                                      | 32          |
| 7    | notification                                 | 0           |
+------+----------------------------------------------+-------------+

[14:05:57]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  `download_dsyms` fails to use session from spaceauth
    https://github.com/fastlane/fastlane/issues/18763 [closed] 2 💬
    13 Sep 2021

➡️  match permission denied
    https://github.com/fastlane/fastlane/issues/17810 [closed] 11 💬
    26 Apr 2021

➡️  Exception type: Spaceship::UnauthorizedAccessError
    https://github.com/fastlane/fastlane/issues/16510 [closed] 35 💬
    17 Sep 2020

and 21 more at: https://github.com/fastlane/fastlane/search?q=The%20request%20could%20not%20be%20completed%20because%3A%0A%09Unauthorized%20Access&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.

[!] The request could not be completed because:
	Unauthorized Access

--------------------------------------
I really hope somebody is able to help, 5 years using fastlane this is the first time this happens.

Best regards
Morten

**Comment by alanflickgames** [comment]
I am seeing exactly the same behaviour, when calling
`fastlane deliver`

On my first attempt today, it got as far as the 2FA step and sent me an SMS. On entering that it failed with the "Unauthorized Access" error.  
The second attempt didn't even get that far, it showed "Login to App Store Connect" and then failed immediately.

It's worth noting that AppStore connect is being extremely slow for me even in a browser, so could be a problem at their end.

**Comment by Mathias-CA** [comment]
Same issue since yesterday.

**Comment by plamenzhl** [comment]
Same issue :(

**Comment by TopIvanAbramov** [comment]
Same issue..

**Comment by Hatoine** [comment]
Same issue :((

**Comment by NiklasMencke** [comment]
Same issue... seems to be something new that effects quite a lot of people

**Comment by assembleMHN** [comment]
> Same issue... seems to be something new that is effecting quite a lot of people.
> 
> I tried resetting my password and login in again via 2FA and I have the same problem (even when choosing SMS 2FA)

Im just glad that its not some obscure settings or something i did accidental on my machine, these things can be hard to debug and fix. So  even though its annoying im glad that we are several in the same boat :D

**Comment by TMcLoone** [comment]
I'm also having this issue since last night, tried deleting the spaceship/email folder to reset 2FA and tried using SMS as a fallback with no luck. 

Running with --verbose gives the full error of: 

Auth error received: 'Spaceship::UnauthorizedAccessError', 'Unauthorized Access'. Login in again then retrying after 3 seconds

It tries a few more times and then fails

**Comment by michalsrutek** [comment]
Same issue for couple of hours now. 

Tried cleaning email in spaceship's folder and re-authenticating. Didn't help.

Was able to upload it through Xcode's Organizer, though.

[Apple's Developer Status Page](https://developer.apple.com/system-status/) shows everything's green, as always...

**Comment by assembleMHN** [comment]
I just tried to uninstall and reinstall a older version suggested in another thread:

sudo gem uninstall fastlane
sudo gem install fastlane --version 2.176.0

This worked for me....

**Comment by panderman** [comment]
Maybe your server IP is blocked by Apple, try change ip if you can

**Comment by trungtinh961** [comment]
same issue

**Comment by appinteractive** [comment]
Seems like github has some trouble

```
> git push origin feature/fix-ads:feature/fix-ads
remote: warning: See http://git.io/iEPt8g for more information.        
remote: warning: File ios/Runner.ipa is 87.03 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB        
remote: warning: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.        
remote: Internal Server Error
To github.com:nof/mobile-app.git
 ! [remote rejected] feature/fix-ads -> feature/fix-ads (Internal Server Error)
error: failed to push some refs to 'github.com:nof/mobile-app.git'
```

**Comment by MadsBuus** [comment]
Also ended up downgrading in  Gemfile:

`gem "fastlane", "2.176.0"`

This worked for me

**Comment by jan-happy** [comment]
`gem install fastlane -v 2.176.0`
gives me
`ERROR:  While executing gem ... (NoMethodError)
    undefined method request' for nil:NilClass`
😢

**Comment by techhowdy** [comment]
> Also ended up downgrading in Gemfile:
> 
> `gem "fastlane", "2.176.0"`
> 
> This worked for me

Yeah I was able to overcome the UnAuthorized Access Error - Thank you

**Comment by pacosta613** [comment]
> Also ended up downgrading in Gemfile:
> 
> `gem "fastlane", "2.176.0"`
> 
> This worked for me

This didn't work for me.

**Comment by NilsBerni** [comment]
> > > I just tried to uninstall and reinstall a older version suggested in another thread:
> > > sudo gem uninstall fastlane sudo gem install fastlane --version 2.176.0
> > > This worked for me....
> > 
> > 
> > That didn't work for me. Then apple complained that the given app with the app identifier does not exist.
> > Does anyone have a different workaround?
> 
> I also get the `Could not find app with app identifier [bundle-identifier] in your App Store Connect account` error, after I have downgraded fastlane.

Make sure your app_identifier("your bundle id") is correct in the Appfile

**Comment by vanvo** [comment]
> Also ended up downgrading in Gemfile:
> 
> `gem "fastlane", "2.176.0"`
> 
> This worked for me

Thanks a lot
`--- Successfully uploaded package to App Store Connect. It might take a few minutes until it's visible online. ---`

**Comment by ronpelayo** [comment]
> 



> `gem install fastlane -v 2.176.0` gives me `ERROR: While executing gem ... (NoMethodError) undefined method request' for nil:NilClass` 😢

make sure you have ruby 2.7.4

**Comment by Hatoine** [comment]
Worked for me too, thanks for the workaround

**Comment by vanvo** [comment]
> `gem install fastlane -v 2.176.0` gives me `ERROR: While executing gem ... (NoMethodError) undefined method request' for nil:NilClass` 😢

If it can help you (https://rvm.io/)
```
$ curl -sSL https://get.rvm.io | bash -s stable
(Follow the instructions after the setup is complete ↑↑↑)
$ rvm install ruby-2.7.2
$ rvm use ruby-2.7.2
$ rvm --default use ruby-2.7.2
$ gem install fastlane -v 2.176.0
```

**Comment by jpirovano** [comment]
same issue :/

**Comment by emmanuelajaero** [comment]
> `gem install fastlane -v 2.176.0` gives me `ERROR: While executing gem ... (NoMethodError) undefined method request' for nil:NilClass` 😢

after 
$sudo gem uninstall fastlane
and
$sudo gem install fastlane --version 2.176.0
verify the version of $fastlane -v
 work for me

**Comment by joshdholtz** [comment]
✨ Official Update

This appears to be a duplicate of #20071. Going to close this so we can keep everything together in one issue 🙃

**Comment by Jokotoye18** [comment]
> I just tried to uninstall and reinstall a older version suggested in another thread:
> 
> sudo gem uninstall fastlane sudo gem install fastlane --version 2.176.0
> 
> This worked for me....

**Comment by waleed-sofstica** [comment]
I solved the issue by following steps:
1) Updating the fastlane to latest version (2.205.1 in my case)

    run this command to update fastlane : `bundle update fastlane`

2) signing in to my apple account from my browser. 
3) Creating app specific password from https://appleid.apple.com/account/manage and adding it to fastFile like this:

default_platform(:ios)

    platform :ios do
      ENV["FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD"] = "YOUR-APP-SPECIFIC_PASS"
      desc "Push a new beta build to TestFlight"
      lane :beta do
        increment_build_number(xcodeproj: "Runner.xcodeproj")
        build_app(workspace: "Runner.xcworkspace", scheme: "Runner")
        upload_to_testflight(skip_waiting_for_build_processing: true)
        end
    end

4) running : `fastlane ios beta` to upload my app to appStore

**Comment by Seunope** [comment]
My Fastlane version was 2.204.1 Updating to the latest version (2.205.1) of Fastlane solved the problem for me. I ran the following command 
`sudo gem install fastlane`
`gem cleanup `

**Comment by MorganGarciaBrown** [comment]
`bundle update fastlane` seemed to do the trick for me.

**Comment by ugglr** [comment]
updating to latest worked for me `sudo gem install fastlane`

**Comment by ekbib** [comment]
Just recently got this error on 2.205.1 and after update to 2.205.2 I am getting the same error:
![Screen Shot 2022-04-27 at 3 37 36 PM](https://user-images.githubusercontent.com/104528090/165610771-fe47a5e7-be9a-423f-bbbc-71b90a6e51d1.png)

**Comment by darkbringer1** [comment]
I still have the same issue, I got the error on version 2.205.1 and it still persists in version 2.205.2...
![Screen Shot 2022-04-28 at 13 13 29](https://user-images.githubusercontent.com/40922278/165730944-c983dc8b-91fc-4510-bccd-18c3a809c826.png)
